pxtblocks/blocklydiff.ts

```diff
@@ -401,4 +422,75 @@ namespace pxt.blocks {
         for (const child of Util.toArray(el.children))
             visDom(child, f);
     }
+
+    export function decompiledDiffAsync(oldTs: string, oldResp: pxtc.CompileResult, newTs: string, newResp: pxtc.CompileResult, options: DiffOptions = {}): DiffResult {
+        const log = logger();
+
+        const oldXml = oldResp.outfiles['main.blocks'];
+        let newXml = newResp.outfiles['main.blocks'];
+        log(oldXml);
+        log(newXml);
+
+        // compute diff of typescript sources
+        const diffLines = pxt.diff.compute(oldTs, newTs, {
+            ignoreWhitespace: true,
+            full: true
+        });
+        log(diffLines);
+
+        // build old -> new lines mapping
+        let oldLineStart = 0;
+        let newLineStart = 0;
+        diffLines.forEach((ln, index) => {
+            // moving cursors
+            const marker = ln[0];
+            const line = ln.substr(2);
+            let lineLength = line.length;
+            switch (marker) {
+                case "-": // removed
+                    oldLineStart += lineLength + 1;
+                    break;
+                case "+": // added
+                    newLineStart += lineLength + 1;
+                    break;
+                default: // unchanged
+                    // skip leading white space
+                    const lw = /^\s+/.exec(line);
+                    if (lw) {
+                        const lwl = lw[0].length;
+                        oldLineStart += lwl;
+                        newLineStart += lwl;
+                        lineLength -= lwl;
+                    }
+                    // find block ids mapped to the ranges
+                    const oldid = pxt.blocks.findBlockId(oldResp.sourceMap, {
+                        start: oldLineStart,
+                        length: lineLength
+                    });
+                    const newid = pxt.blocks.findBlockId(newResp.sourceMap, {
+                        start: newLineStart,
+                        length: lineLength
+                    });
```

Comment:
After my PR, you can provide a position within a single line, in case that helps

Reply:
right now i don't handle line diff but eventually we could be more precise and flag changes within nested blocs

Reply:
or after my PR, you can update my decompiler. why do we need this?

Reply:
I think the use case is for tutorial steps that just modify a line like "change this number" or "add (pick random) into the second argument". Not sure how many non-trivial examples of this we have.